parkbenchcommon/ramdisk.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2018 Joel Allen Luellwitz and Emily Frost
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Provides a class for managing ramdisks."""
+
+__all__ = ['Ramdisk', 'RamdiskMountError', 'RamdiskOptionError']
+
+import logging
+import os
+import subprocess
+
+class RamdiskMountError(Exception):
+    """Raised when a ramdisk mount operation fails."""
+
+class RamdiskOptionError(ValueError):
+    """Raised when an option for a ramdisk mount is invalid."""
+
+
+class Ramdisk:
+    """A class for managing ramdisks."""
+
+    def __init__(self, path):
+        """Stores mountpoint and options.
+
+        path: A string indicating the path where the ramdisk will be mounted.
+        """
+        self.logger = logging.getLogger(__name__)
+
+        # The mount command outputs the canonical path of all mountpoints, so it will need
+        #   to know the absolute path with any links resolved. Python calls this realpath.
+        self.path = os.path.realpath(path)
+
+    def mount(self, size, uid, gid, mode):
+        """Mounts the ramdisk. Raises an exception on failure, and does nothing if the disk
+        is already mounted.
+
+        size: The size of the ramdisk to be mounted. This should be a string representing a
+          number of bytes, and may include the single-character suffixes k, m, g, or % for
+          kibibytes, mebibytes, gibibytes, or percentage of physical RAM, respectively.
+        uid: The system user ID that should own the mount directory.
+        gid: The system group ID that should be associated with the mount directory.
+        mode: The mode of the mount directory access permissions. This should be a decimal
+          integer. ex: 511, not 777.
+          Note: This integer is most easily obtained by ORing the appropriate permission
+          flags from the stat module.
+        """
+
+        # Since size should be an integer with a single character suffix, we validate only
+        #   the integer part by including all but the last character of the string.
+        self._validate_integer_option('size', size[:-1])
```

Comment:
We should still verify the last character is a single byte, lowercase, alpha character (or percent). Also, what if the size is just the number of bytes with no suffix? (We could just declare an option without a suffix as invalid.)

Reply:
There are actually a small enough number of valid options to keep them in a dict.